tests/all/test_attributes.rs

```diff
@@ -40,6 +40,36 @@ fn test_string_attributes() {
     assert_eq!(string_attribute.get_string_value().to_str(), Ok("my_val"));
 }
 
+#[llvm_versions(12.0..=latest)]
+#[test]
+fn test_type_attribute() {
+    use inkwell::types::AnyType;
+
+    let context = Context::create();
+    let kind_id = Attribute::get_named_enum_kind_id("sret");
+    let any_type = context.i32_type().as_any_type_enum();
+    let type_attribute = context.create_type_attribute(
+        kind_id,
+        any_type,
+    );
+
+    assert!(type_attribute.is_type());
+    assert_eq!(type_attribute.get_type_value(), any_type);
+    assert_ne!(type_attribute.get_type_value(), context.i64_type().as_any_type_enum());
+
+    let context = Context::create();
+    let kind_id = Attribute::get_named_enum_kind_id("byval");
+    let any_type = context.i128_type().as_any_type_enum();
+    let type_attribute = context.create_type_attribute(
+        kind_id,
+        any_type,
+    );
+
+    assert!(type_attribute.is_type());
```

Comment:
Could we run these tests against more types to verify any value of AnyTypeEnum is valid as an attribute? Maybe in a loop of types? Like

```rust
let types = [int_type.into_any_type(), float_type.into(), vec_type.into(), ...];

for type in types {
    // create_type_attr
    // asserts
    // ...
}
```

Reply:
It works for all `AnyTypeEnum` variants as tested in newer commit.